logger/logrus.go

```diff
@@ -30,48 +30,52 @@ func (logrus logrusLogger) Prettyf(msg string, obj interface{}) {
 }
 
 func (logrus logrusLogger) Errorf(format string, args ...interface{}) {
-	logrus.Logger.Errorf(format, args...)
+	logrus.FieldLogger.Errorf(format, args...)
 }
 
 func (logrus logrusLogger) Debugf(format string, args ...interface{}) {
-	logrus.Logger.Debugf(format, args...)
+	logrus.FieldLogger.Debugf(format, args...)
 }
 
 func (logrus logrusLogger) Tracef(format string, args ...interface{}) {
-	logrus.Logger.Tracef(format, args...)
+	logrus.FieldLogger.Tracef(format, args...)
 }
 
 func (logrus logrusLogger) Fatalf(format string, args ...interface{}) {
-	logrus.Logger.Fatalf(format, args...)
+	logrus.FieldLogger.Fatalf(format, args...)
 }
 
 func (logrus logrusLogger) NewLogger(key string, value interface{}) Logger {
-	return logrusLogger{Logger: logrusapi.New().WithField(key, value).Logger}
+	return logrusLogger{FieldLogger: logrusapi.New().WithField(key, value)}
 }
 
 func (logrus logrusLogger) NewLoggerWithFields(fields map[string]interface{}) Logger {
-	return logrusLogger{Logger: logrusapi.New().WithFields(logrusapi.Fields(fields)).Logger}
+	return logrusLogger{FieldLogger: logrusapi.New().WithFields(logrusapi.Fields(fields))}
+}
+
+func (logrus logrusLogger) WithFields(fields map[string]interface{}) Logger {
+	return logrusLogger{FieldLogger: logrus.FieldLogger.WithFields(logrusapi.Fields(fields))}
```

Comment:
WithFields shouldn't be necessary, as NewLogger is called on logsrusLogger it should create a sub logger via WithField(), not a brand new logger

Reply:
NewLogger explicity creates  a new base logger each time.  WithFields is intended to retain any other fields that may previously have been added.